app/Models/User.php

```diff
@@ -1373,19 +1373,34 @@ public function visibleGroups()
                 return [app('groups')->byIdentifier('bot')];
             }
 
-            $ids = $this->groupIds();
-            array_unshift($ids, $this->defaultGroup()->getKey());
-
             $groups = [];
-            foreach (array_unique($ids) as $id) {
-                $group = app('groups')->byId($id);
+            foreach ($this->userGroups as $userGroup) {
+                $cachedGroup = app('groups')->byId($userGroup->group_id);
+                if (!$cachedGroup) {
+                    continue;
+                }
+
+                $group = clone $cachedGroup;
+
+                if ($group && $group->has_playmodes) {
```

Comment:
the first [truthy](https://www.php.net/manual/en/language.types.boolean.php#language.types.boolean.casting) check is rather useless. Similarly the `optional` wrapper below. Additionally there's no need to `clone` if `display_order` is `null`.